Gems/Terrain/Code/Tests/TerrainSystemTest.cpp

```diff
@@ -572,4 +594,308 @@ namespace UnitTest
+
+            { AZ::Vector2(-3.25f, -5.25f), -8.5f }, // Should return a height of -3.25 + -5.25
+            { AZ::Vector2(-7.71f, -9.74f), -17.45f }, // Should return a height of -7.71 + -9.74
+        };
+
+        auto perPositionCallback = [&testPoints](const AZ::Vector3& inPosition, float height, bool terrainExists){
+            bool found = false;
+            for (auto& testPoint : testPoints)
+            {
+                if (testPoint.m_testLocation.GetX() == inPosition.GetX() && testPoint.m_testLocation.GetY() == inPosition.GetY())
+                {
+                    constexpr float epsilon = 0.0001f;
+                    EXPECT_NEAR(height, testPoint.m_expectedHeight, epsilon);
+                    found = true;
+                    break;
+                }
+            }
+            EXPECT_EQ(found, true);
+        };
+
+        AZStd::vector<AZ::Vector3> inPositions;
+        for (auto& testPoint : testPoints)
+        {
+            AZ::Vector3 position(testPoint.m_testLocation.GetX(), testPoint.m_testLocation.GetY(), 0.0f);
+            inPositions.push_back(position);
+        }
+
+        terrainSystem->ProcessHeightsFromList(inPositions, perPositionCallback, AzFramework::Terrain::TerrainDataRequests::Sampler::BILINEAR);
+    }
+
+    TEST_F(TerrainSystemTest, TerrainProcessNormalsFromListWithBilinearSamplers)
+    {
+        // Similar to TerrainProcessHeightsFromListWithBilinearSamplers but for normals
+
+        const AZ::Aabb spawnerBox = AZ::Aabb::CreateFromMinMaxValues(-10.0f, -10.0f, -5.0f, 10.0f, 10.0f, 15.0f);
+        const float amplitudeMeters = 10.0f;
+        const float frequencyMeters = 1.0f;
+        auto entity = CreateAndActivateMockTerrainLayerSpawner(
+            spawnerBox,
+            [amplitudeMeters, frequencyMeters](AZ::Vector3& position, bool& terrainExists)
+            {
+                // Our generated height will be X + Y.
+                float expectedHeight = position.GetX() + position.GetY();
+
+                // If either X or Y aren't evenly divisible by the query frequency, add a scaled value to our generated height.
+                // This will show up as an unexpected height "spike" if it gets used in any bilinear filter queries.
+                float unexpectedVariance =
+                    amplitudeMeters * (fmodf(position.GetX(), frequencyMeters) + fmodf(position.GetY(), frequencyMeters));
+                position.SetZ(expectedHeight + unexpectedVariance);
+                terrainExists = true;
+            });
+
+        // Create and activate the terrain system with our testing defaults for world bounds, and a query resolution at 1 meter intervals.
+        const AZ::Vector2 queryResolution(frequencyMeters);
+        auto terrainSystem = CreateAndActivateTerrainSystem(queryResolution);
+
+        const NormalTestPoint testPoints[] = {
+
+            { AZ::Vector2(0.0f, 0.0f), AZ::Vector3(-0.5773f, -0.5773f, 0.5773f) },
```

Comment:
The expected values for normals were plugged in from the normals generated by calling the GetNormal() function in a loop. This is still using the same input values and height gradient lambda as the ProcessHeightsList test. Maybe come up with better test inputs. Until then this works as a placeholder.

Reply:
Yeah, I still have it in my backlog to finish off unit tests for the GetNormal calls.  Open to suggestions if you have any ideas for good ways to test it.  When I started writing the tests, I realized I'm not sure what the "correct" values for the normals should actually be, which makes it hard to test them. :(